shared/js/content-scripts/element-hiding.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * The purpose of this content script is to hide third-party tracking ads that have been blocked
+ * and appear as blank spaces on websites. In order to distinguish between tracking ads and legitimate
+ * ads, it is necessary to inject this script into all frames.
+ */
+
+var contentScript = {}
+contentScript.frameType = window === window.top ? 'main' : window.parent === window.top ? 'topLevelFrame' : 'nestedFrame'
+contentScript.possibleTargets = []
+contentScript.foundScripts = []
+contentScript.foundFrames = []
+
+/**
+ * Set up listener in each frame to respond to messages from other frames.
+ * There are three types of messages being passed:
+ * 1. iframes send messages to the main frame requesting their id (frameIdRequest)
+ * 2. main frame sends messages back to iframes with their id (setFrameId)
+ * 3. iframes send messages to main frame when they contain blocked elements (hideFrame)
+ */
+contentScript.frameListener = (e) => {
+    if (e.data.type === 'frameIdRequest') {
+        contentScript.foundFrames = document.getElementsByTagName('iframe')
```

Comment:
I think `<iframe>` is not the only "frame object" - there is also `<frame>`, `<object>` and `<embed>` that (as far as I remember) can be used similarly. We can get all of those with `window.frames`.

Reply:
I think for now, just for the sake of timeliness, we only care about iframes. Good catch though, we can address in a future PR.